RadarSDK/RadarLocationPremissionManager.m

```diff
@@ -0,0 +1,190 @@
+//
+//  RadarLocationPremissionsManager.m
+//  RadarSDK
+//
+//  Created by Kenny Hu on 5/8/24.
+//  Copyright © 2024 Radar Labs, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "RadarLocationPermissionManager.h"
+#import "RadarLocationPermissionStatus+Internal.h"
+#import "RadarDelegateHolder.h"
+
+@interface RadarLocationPermissionManager ()
+
+@property (assign, nonatomic) BOOL danglingBackgroundPermissionsRequest;
+@property (assign, nonatomic) BOOL inBackgroundLocationPopUp;
+
+@end
+
+@implementation RadarLocationPermissionManager
+
++ (instancetype)sharedInstance {
+    static dispatch_once_t once;
+    static id sharedInstance;
+    dispatch_once(&once, ^{
+        sharedInstance = [self new];
+    });
+    return sharedInstance;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.locationManager = [CLLocationManager new];
+        self.locationManager.delegate = self;
+        RadarLocationPermissionStatus *status = [RadarLocationPermissionStatus getRadarLocationPermissionStatus];
+        if (status) {
+            self.status = status;
+            // we should not start in the popup state
+            self.status.inForegroundPopup = NO;
+        } else{
+            if (@available(iOS 14.0, *)) {
+                self.status = [[RadarLocationPermissionStatus alloc] initWithStatus:self.locationManager.authorizationStatus
+                                                            backgroundPopupAvailable:YES
+                                                                   inForegroundPopup:NO
+                                                   userRejectedBackgroundPermissions:NO];
+            } else {
+                // just a dummy value, this component will not communicate with the rest of the SDK if the version is below 14.0
+                self.status = [[RadarLocationPermissionStatus alloc] initWithStatus: kCLAuthorizationStatusAuthorizedAlways
+                                                            backgroundPopupAvailable:NO
+                                                                   inForegroundPopup:NO
+                                                   userRejectedBackgroundPermissions:NO];
```

Comment:
Thinking ahead to the analytics, is this the right way of create a dummy value for when it's `unknown`?

Reply:
I'm just going to make this a nullable property and leave it as null if the iOS version is too low.